### PR TITLE
checker: check fn decl for anon fns

### PR DESCRIPTION
**Pull request by Delta456**
Fixes https://github.com/vlang/v/issues/7482

```
foo.v:21:24: error: missing return at end of function `anon_441_7_16`
   19 | }
   20 | */
   21 | println([1,2,3].filter(fn (n int) bool {
      |                        ~~
   22 |     true
   23 | }))
   ```

### CODE REVIEW

File: vlib/v/checker/tests/fn_var.out [hunk 1]
```diff
@@ -1,3 +1,8 @@
+vlib/v/checker/tests/fn_var.vv:1:10: error: missing return at end of function `anon_27_7_9`
```
Comment (Delta456) [inline vlib/v/checker/tests/fn_var.out:1]: I don't like the name `anon_27_7_9`. Can anyone suggest a better way to use the proper naming for `ast.AnonFn`'s name. I know we have to use `typ` field of it but how will I pass it through?

**Comment by Delta456** [comment]
Gitly fails due to an autofree bug.

File: vlib/v/checker/tests/fn_var.out (hunk 1 above)
Comment (spytheman) [reply vlib/v/checker/tests/fn_var.out:1]: just `anonymous function` ?

Comment (spytheman) [reply vlib/v/checker/tests/fn_var.out:1]: `error: missing return at the end of an anonymous function`

Comment (Delta456) [reply vlib/v/checker/tests/fn_var.out:1]: I thought of doing that but it will look inconsistent with other functions.

**Comment by medvednikov** [comment]
I'm working on fixing the -autofree bug.

File: vlib/v/checker/tests/fn_var.out (hunk 1 above)
Comment (medvednikov) [reply vlib/v/checker/tests/fn_var.out:1]: `error: missing return at the end of an anonymous function` is fine

Comment (Delta456) [reply vlib/v/checker/tests/fn_var.out:1]: Ok

Comment (danieldaeschle) [reply vlib/v/checker/tests/fn_var.out:1]: i think we have something that makes `fn (v int) bool` possible (the function signature).

Comment (Delta456) [reply vlib/v/checker/tests/fn_var.out:1]: @danieldaeschle how will this be done in without `table.type`?

**Pull request merged**